Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -1106,8 +1106,8 @@ bool WebPageProxy::suspendCurrentPageIfPossible(API::Navigation& navigation, Ref
     if (!mainFrame)
         return false;
 
-    if (openerFrame() && (preferences().processSwapOnCrossSiteWindowOpenEnabled() || preferences().siteIsolationEnabled())) {
-        WEBPAGEPROXY_RELEASE_LOG(ProcessSwapping, "suspendCurrentPageIfPossible: Not suspending current page for process pid %i because it has an opener.", m_process->processID());
+    if (preferences().processSwapOnCrossSiteWindowOpenEnabled() || preferences().siteIsolationEnabled()) {
```

Comment:
I don't think this is sufficient. I think you also need to disable the backforward cache at WebCore-level, no?
We we navigate from origin A to origin A, WebCore will put that first page in the cache and the WebPageProxy logic here won't be involved since no PSON. Yet, that first page may have cross-origin frames (and thus be broken with full site isolation).